src/common/Types.ts

```diff
@@ -732,6 +732,7 @@ export interface GestureViewProps extends CommonStyledProps<ViewStyleRuleSet>, C
     onPanHorizontal?: (gestureState: PanGestureState) => void;
     onTap?: (gestureState: TapGestureState) => void;
     onDoubleTap?: (gestureState: TapGestureState) => void;
+    onContextMenuGesture?: (gestureState: TapGestureState) => void;
```

Comment:
This should be "onContextMenu" for consistency. We don't add the word "Gesture" in any other cases.

Reply:
This can't be `onContextMenu` because of conflicting interface. As discussed in the chat you created:

> `onContextMenu` is defined as `(e: MouseEvent)` which 1) doesn't make sense for touch interface 2) would be hard to fake due to how GestureView works now and would require serious refactoring of the component.
> 
> All the other callbacks for GestureView are `(gestureState: TapGestureState)` so my proposal is to keep up with this and add `onContextMenuGesture(gestureState: TapGestureState).` Thoughts?

Reply:
Yeah, I realize I used the name "onContextMenuGesture" in the chat, but now that I'm reviewing the change, I think "onContextMenu" is better for consistency. There's no conflict with ViewPropsShared because GestureProps has no inheritance relationship with ViewPropsShared.

The GestureView documentation also needs to be updated to include the new prop.

Reply:
Yes, good point about documentation. I'll add it.

> There's no conflict with ViewPropsShared because GestureProps has no inheritance relationship with ViewPropsShared.

src/native-common/GestureView.tsx(42,52): error TS2344: Type 'GestureViewProps' does not satisfy the constraint 'ViewProps'.
  Types of property 'onContextMenu' are incompatible.
    Type '((gestureState: TapGestureState) => void) | undefined' is not assignable to type '((e: MouseEvent) => void) | undefined'.

`GestureView extends ViewBase<Types.GestureViewProps, {}>` -> `ViewBase<P extends Types.ViewProps, S>` -> `ViewProps extends ViewPropsShared`. Therefore `GestureViewProps` need to be interface compatible with `ViewPropsShared`.

Reply:
Yikes, that's a bug. GestureView shouldn't be extending ViewBase like that. Neither should ScrollView. I'm surprised that the TypeScript compiler didn't complain about that because ViewBase requires that "P extends Types.ViewProps", and GestureViewProps doesn't properly extend this interface. I'm working on a fix, and I'll let you know when I'm done.

Reply:
OK, I just pushed a fix for the bug. Turns out it was a problem for some implementations of GestureView, ScrollView and WebView. I've fixed all of them.